nacl-fast.js

```diff
@@ -2161,8 +2183,16 @@ function checkBoxLengths(pk, sk) {
 
 function checkArrayTypes() {
   for (var i = 0; i < arguments.length; i++) {
-    if (!(arguments[i] instanceof Uint8Array))
-      throw new TypeError('unexpected type, use Uint8Array');
+    if (arguments[i] instanceof Object && typeof arguments[i].length === 'number') {
```

Comment:
All that the implementation demands of the input is that:

* it be indexable
* it have a `length` property

You don't have to be a `Uint8Array` to offer those two guarantees.